engine/src/main/java/org/destinationsol/game/FactionInfo.java

```diff
@@ -40,32 +34,26 @@ public FactionInfo() {
     }
 
     private void createFactionList() {
-        String[] moduleList = getModuleList();
-        for(String module: moduleList) {
-            String path =  module + ":factions";
-            System.out.println(path.equals("case:factions"));
-            System.out.println("core:factions");
-            Json factionJSON = Assets.getJson("core:factions");
-            for(int n = 0; n < factionJSON.getJsonValue().getJSONArray("factions").length(); n++) {
-                factionName.add(factionJSON.getJsonValue().getJSONArray("factions").getJSONObject(n).getString("name").replace("\"", ""));
-                factionColor.add(factionJSON.getJsonValue().getJSONArray("factions").getJSONObject(n).getString("color").replace("\"", ""));
-                factionDisposition.add(factionJSON.getJsonValue().getJSONArray("factions").getJSONObject(n).getInt("disposition"));
+        for (String module : getModuleList()) {
+            String path = module.replaceAll("[[, ]]", "") + ":factions";
+            Json factionJson = Assets.getJson(path);
+            for (int n = 0; n < factionJson.getJsonValue().getJSONArray("factions").length(); n++) {
```

Comment:
Please extract `factionJson.getJsonValue().getJSONArray("factions")` into a variable to improve readability